src/borg/archive.py

```diff
@@ -1431,19 +1464,22 @@ def process_file(self, *, path, parent_fd, name, st, cache, flags=flags_normal):
                     else:
                         status = "M" if known else "A"  # regular file, modified or added
                     self.print_file_status(status, path)
+                    self.stats.files_stats[status] += 1
                     status = None  # we already printed the status
                     # Only chunkify the file if needed
                     if chunks is not None:
                         item.chunks = chunks
                     else:
                         with backup_io("read"):
+                            started_chunking = time.monotonic()
                             self.process_file_chunks(
                                 item,
                                 cache,
                                 self.stats,
                                 self.show_progress,
                                 backup_io_iter(self.chunker.chunkify(None, fd)),
                             )
+                            self.stats.chunking_time += time.monotonic() - started_chunking
```

Comment:
Hmm, guess this measures way more than just the chunking time:

it also measures the time to do **all processing** for each chunk, right?

guess we should rather only measure **reading and chunking** (== cutting into pieces) of the input data for this stats value (separating reading from chunking might be hard, the buzhash chunker is C code and does reading via chunking buffer refills, so guess we just measure both together for simplicity).

but after chunks are cut, they get:
- hashed (id_hash) - you already have that (and this is interesting because it might be slow on some CPUs)
- compressed (very interesting, this can be very fast or very slow, depending on algorithm and level)
- encrypted (very interesting, can be very fast or very slow, depending on cpu and algorithm)
- (transmitted and) written to repo (also very interesting, can be slow for slow remote repos)

Reply:
for this PR, maybe you could fix the chunking_time to really only measure chunking/reading.

but then I guess it would be good to merge this PR before working on the other items to not widen the scope too much, we already have quite a lot of commits here.

Reply:
So, do I have to measure only the call of `chunkify`? Line 1481 there.

Reply:
That's not a simple call, but that returns an iterator. Have a look into `chunker.pyx -> ChunkerFixed`.

Reply:
`*.pyx` is cython code and needs recompiling after changes, easiest is something like:
```
python setup.py clean clean2 ; pip install -e .
```

Reply:
As I understand, the chunking is done lazily at `Chunker.__next__` and `ChunkerMixed.chunkify` (a generator), so I should measure these methods.
This is easy to do, but these methods don't have access to the Statistics object.
I think I could add the time it took to read the chunk as an attribute to the Chunk class, so the chunker methods could yield/return like this: `yield Chunk(data, size=got, allocation=allocation, chunking_time=measured_time)`.
And the process_file_chunks would look something like this:
```python
for chunk in chunk_iter:
            stats.chunking_time += chunk.chunking_time
            item.chunks.append(chunk_processor(chunk))
...
```

Reply:
Guess you could either give stats to the chunker init or have the timing in the chunker instance and pick it from there from the calling code.

We're only interested in the overall chunking time, so better do not put it into the Chunk instance, but into the Chunker instance.

Reply:
Am I gonna need to modify pyx and c files to do that?

Reply:
Likely the chunker.pyx, just have a look.